controllers/profiles/preview/states_preview.go

```diff
@@ -62,6 +64,25 @@ func (h *newBuilderState) Do(ctx context.Context, workflow *operatorapi.SonataFl
 		klog.V(log.E).ErrorS(err, "Failed to get active platform")
 		return ctrl.Result{RequeueAfter: requeueWhileWaitForPlatform}, nil, err
 	}
+
+	//TODO WM remove this comment.
+	fmt.Printf("XXXX: Preparing workflow properties before build for workflow: %s\n", workflow.Name)
+	// Ensure the properties are prepared before starting the build process, and thus, we make them available at build time.
+	userPropsCM, _, err := h.ensurers.userPropsConfigMap.Ensure(ctx, workflow)
```

Comment:
@ricardozanini And here, to ensure the configmaps with the applications properites are available at build time, we create them before going to the build phase.